OutOfSchool/OutOfSchool.WebApi.Tests/Extensions/ApplicationExtensionTests.cs

```diff
@@ -0,0 +1,108 @@
+﻿using System.Collections.Generic;
+using System.Linq;
+using NUnit.Framework;
+using OutOfSchool.Services.Enums;
+using OutOfSchool.Services.Models;
+using OutOfSchool.Tests.Common.TestDataGenerators;
+using OutOfSchool.WebApi.Extensions;
+
+namespace OutOfSchool.WebApi.Tests.Extensions;
+
+[TestFixture]
+public class ApplicationExtensionTests
+{
+    [Test]
+    public void ApplicationExtensions_AmountOfPendingApplications_ReturnValue()
+    {
+        // Arrange
+        var applications = GetApplications();
+
+        // Act
+        var result = applications.AmountOfPendingApplications();
+        var expected = applications.Count(x =>
+            x.Status == ApplicationStatus.Pending
+            && !x.IsDeleted
+            && (x.Child == null || !x.Child.IsDeleted)
+            && (x.Parent == null || !x.Parent.IsDeleted));
+
+        // Assert
+        Assert.AreEqual(expected, result);
+    }
+
+    [Test]
+    public void ApplicationExtensions_TakenSeats_ReturnValue()
+    {
+        // Arrange
+        var applications = GetApplications();
+
+        // Act
+        var result = applications.TakenSeats();
+        var expected = applications.Count(x =>
+            (x.Status == ApplicationStatus.Approved || x.Status == ApplicationStatus.StudyingForYears)
+            && !x.IsDeleted
+            && (x.Child == null || !x.Child.IsDeleted)
+            && (x.Parent == null || !x.Parent.IsDeleted));
+
+        // Assert
+        Assert.AreEqual(expected, result);
+    }
+
+    private List<Application> GetApplications()
```

Comment:
(Pitfalls of separating logic from the test)
It's common approach to reuse the common code by extracting it to helper methods, even in testing, but this method has more than one responsibility and hides the complexity.

I.e., it looks like `child == null` or `parent == null` conditions are never addressed by the generated data, so never asserted in the related tests. Please reconsider and (maybe) refactor it.

Reply:
Deleted it.